code/modules/awaymissions/zlevel.dm

```diff
@@ -12,7 +12,7 @@ GLOBAL_LIST_INIT(potentialRandomZlevels, generateMapList(filename = "[global.con
 		to_chat(world, "<span class='boldannounce'>Away mission loaded.</span>")
 
 /proc/reset_gateway_spawns(reset = FALSE)
-	for(var/obj/machinery/gateway/G in world)
+	for(var/obj/machinery/gateway/G in GLOB.gateways)
```

Comment:
You can just use GLOB.machines, no reason to make another global list.

Reply:
yes, thanks